crates/swc_cli/src/commands/plugin.rs

```diff
@@ -0,0 +1,234 @@
+///
+/// Note: this is slim implementation does not support different vcs other than
+/// git at the moment.
+fn write_ignore_file(base_path: &Path) -> Result<()> {
+    let ignore_list: Vec<String> = vec!["/target", "^target/", "target"]
+        .iter()
+        .map(|v| v.to_string())
+        .collect();
+
+    let ignore_file_path = base_path.join(".gitignore");
+
+    let ignore: String = match paths::open(&ignore_file_path) {
+        Err(err) => match err.downcast_ref::<std::io::Error>() {
+            Some(io_err) if io_err.kind() == ErrorKind::NotFound => ignore_list.join("\n") + "\n",
+            _ => return Err(err),
+        },
+        Ok(file) => {
+            let existing = BufReader::new(file);
+
+            let existing_items = existing.lines().collect::<Result<Vec<_>, _>>().unwrap();
+            let mut out = String::new();
+
+            out.push_str("\n\n# Added by swc\n");
+            if ignore_list.iter().any(|item| existing_items.contains(item)) {
+                out.push_str("#\n# already existing elements were commented out\n");
+            }
+            out.push('\n');
+
+            for item in &ignore_list {
+                if existing_items.contains(item) {
+                    out.push('#');
+                }
+                out.push_str(item);
+                out.push('\n');
+            }
+
+            out
+        }
+    };
+
+    paths::append(&ignore_file_path, ignore.as_bytes())?;
+    Ok(())
+}
+
+impl super::CommandRunner for PluginScaffoldOptions {
+    /// Create a rust project for the plugin from template.
+    /// This largely mimic https://github.com/rust-lang/cargo/blob/master/src/cargo/ops/cargo_new.rs,
+    /// but also thinner implementation based on some assumptions like skipping
+    /// to support non-git based vcs.
+    fn execute(&self) -> Result<()> {
+        let path = &self.path;
+        if path.exists() {
+            anyhow::bail!("destination `{}` already exists", path.display())
+        }
+
+        let name = get_name(self)?;
+
+        // Choose to rely on system's git binary instead of depends on git lib for now
+        // to avoid bring in large / heavy dependencies into cli binaries.
+        // Depends on our usecase grows, we can revisit this.
+        let mut base_git_cmd = if cfg!(target_os = "windows") {
+            let mut c = std::process::Command::new("cmd");
+            c.arg("/C").arg("git");
+            c
+        } else {
+            std::process::Command::new("git")
+        };
+
+        // init git repo
+        base_git_cmd
+            .args(["init", name])
+            .output()
+            .context("failed to create dir for the plugin")?;
+
+        // generate .gitignore
+        write_ignore_file(path)?;
+
+        // Create `Cargo.toml` file with necessary sections
+        paths::write(
+            &path.join("Cargo.toml"),
+            format!(
+                r#"[package]
+name = "{}"
+version = "0.1.0"
+edition = "2018"
```

Comment:
why not use edition 2021 by default

Reply:
No major reason. Unsure if it should be or not?

Reply:
I think 2021 would be better, as swc uses 2021.